Return to resource list after adding/removing an item from a lesson/quiz

MisRob opened the pull request:
### Summary

Redirect coaches to a list after adding or removing an item from a lesson/quiz when planning a class. Display snackbar on a list page.

| Lesson | Quiz |
| --------- | ------- |
![lesson-resource-added](https://user-images.githubusercontent.com/13509191/86032400-9987c180-ba37-11ea-941b-e06f3c7353fb.png) | ![quiz-resource-added](https://user-images.githubusercontent.com/13509191/86032676-056a2a00-ba38-11ea-9633-4f1a457ee909.png) |
| ![lesson-resource-removed](https://user-images.githubusercontent.com/13509191/86032549-d2c03180-ba37-11ea-92e3-8d3714a104ed.png) | ![quiz-resource-removed](https://user-images.githubusercontent.com/13509191/86033008-8de8ca80-ba38-11ea-9754-f08ed253f93a.png) |

### Reviewer guidance

**Lesson**
- Coach -> Plan -> Lessons -> select a lesson/create a new lesson -> MANAGE RESOURCES -> open a resource preview
- Click ADD/REMOVE

**Quiz**
- Coach -> Plan -> Quizzes -> create a new quiz -> open a resource preview
- Click ADD/REMOVE

### References

Closes #6819
Opened #7212

----

### Contributor Checklist


PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

Commit:
toolbarRoute -> returnBackRoute

Commit:
Redirect to a list after adding/removing an item

jonboiser commented:
We should prevent the Preview pages  changing in appearance before navigating back to the resource-list page (see GIF).

Could you

1. Remove the "Added" indicator from appearing
1. Prevent the button label from switching from "Add" <-> "Remove" before redirect happens
1. (Optionally) disable the button to discourage double-clicking the button (even though double-clicking does not appear to cause problems)


![CleanShot 2020-06-29 at 11 11 33](https://user-images.githubusercontent.com/10248067/86040694-59095300-b9f9-11ea-8247-782e3a7280d2.gif)

jonboiser commented on kolibri/plugins/coach/assets/src/views/plan/PlanQuizPreviewPage.vue:
```diff
@@ -55,11 +58,15 @@
         this.addToSelectedExercises([content]);
-        this.createSnackbar(this.$tr('added', { item: this.currentContentNode.title }));
+        this.$router.push(this.returnBackRoute).then(() => {
+          this.createSnackbar(this.$tr('added', { item: this.currentContentNode.title }));
```
You can fix #7212 at this line by adding the `commonCoreString` mixin (which provides a `showSnackbarNotification` method) to the component and replacing the line here

```suggestion
          this.showSnackbarNotification('resourcesAddedWithCount', { count: 1 });
```

jonboiser commented on kolibri/plugins/coach/assets/src/views/plan/PlanQuizPreviewPage.vue:
```diff
@@ -55,11 +58,15 @@
         this.removeFromSelectedExercises([content]);
-        this.createSnackbar(this.$tr('removed', { item: this.currentContentNode.title }));
+        this.$router.push(this.returnBackRoute).then(() => {
+          this.createSnackbar(this.$tr('removed', { item: this.currentContentNode.title }));
```
Similarly,

```suggestion
          this.showSnackbarNotification('resourcesRemovedWithCount', { count: 1 });
```

jonboiser commented:
I also added some suggestions for fixing #7212 inside some comments

Commit:
Select options behavior improvements

- do not switch between added/removed states before redirect
- disable add/remove button after click

Commit:
Consistent snackbars labels

MisRob commented:
Hi JB, thank you. I've just pushed all updates.

Regarding "Added" indicator, although it's not rendered after "Add" button click anymore, it's still displayed on the preview page load - is it fine or do you want to remove it completely? I've tried both options and it seems that keeping indicator for this situation makes visually clear if it's added or not at first glance when compared to displaying "Remove" button only.

MisRob commented:
Also added a commit to unify snackbar messages that closes #7212.

MisRob commented on kolibri/plugins/coach/assets/src/views/plan/LessonContentPreviewPage/index.vue:
```diff
@@ -173,6 +190,11 @@
       },
     },
+    watch: {
+      isSelected() {
```
This watcher is not currently used though I added it because I would expect that `disableSelectButton` will be appropriately reset on selection state change.

jonboiser reviewed (approved):
The changes look great, thanks!

jonboiser committed:
Add hack to prevent undoing the removal of the 1 and only resource

jonboiser commented:
@MisRob I found a bug where if you removed the only resource in a lesson from the preview page, it would be automatically re-added when navigating back to the resource list (due to the complicated autosaving logic we use). I added one more commit as a hack to fix this by committing the resource removal _before_ the navigation, but keeping the Preview page buttons and indicators saying "added" until the navigation occurs.
